### PR TITLE
Test improvements

### PR DESCRIPTION
**Pull request by ltseeley**

### CODE REVIEW

**Comment by vaporos** [comment]
It isn't clear that the changes in the tests accomplish the same thing as the original tests. The Original tests were organized around real-world examples of how one might use the API (or, rather, that was the intent). Its not obvious this has been translated into the replacement tests. It might help to clarify "X is covered by Y" and then enhance the fn doc to explictly state what is being covered by the test.

**Comment by ltseeley** [comment]
> It isn't clear that the changes in the tests accomplish the same thing as the original tests. The Original tests were organized around real-world examples of how one might use the API (or, rather, that was the intent). Its not obvious this has been translated into the replacement tests. It might help to clarify "X is covered by Y" and then enhance the fn doc to explictly state what is being covered by the test.

I updated the commit message to indicate which new tests replace which old ones. I also added a multi-key, single-context test which I realized was missing in the new tests. Let me know if anything seems to need further clarification.

**Review by vaporos — APPROVED** [review]

File: libcylinder/src/key.rs [hunk 1]
```diff
@@ -93,6 +101,14 @@ impl PrivateKey {
     }
 }
 
+impl std::fmt::Debug for PrivateKey {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        f.debug_struct("PrivateKey")
+            .field("bytes", &self.as_hex())
```
Comment (peterschwarz) [inline libcylinder/src/key.rs:107]: Is logging the complete key a security concern? Especially if this is just for testing purposes

Comment (ltseeley) [reply libcylinder/src/key.rs:107]: I guess the debug/display implementations aren't needed for the tests; I can remove them for private key if this is a security issue.

Comment (ltseeley) [reply libcylinder/src/key.rs]: Removed the display/debug impls for private key and updated the commit

**Review by peterschwarz — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**